### PR TITLE
feature added open image context in md preview

### LINKED ISSUE #184553: markdown preview: allow opening images in a new editor
Images opened in the markdown preview could have a context menu that includes 'Open Image' that opens the image in a new editor.

### PR DESCRIPTION
**Pull request by Gobinath-B**
fix #184553

## changes made

- [x] add open image context item in ui md preview
- [ ] functionality for open image in new tab

### CODE REVIEW

**Review by mjbvz — CHANGES_REQUESTED** [review]
Thanks for taking a look

I think this can be simplified a lot by using the [`vscode.open` command](https://code.visualstudio.com/api/references/commands) to show the image. That will open the image using VS Code's default editor. You can call this from the VS Code extension side with the uri of the image

To simplify things, this should only apply to images that are part of the workspace

**Comment by Gobinath-B** [comment]
> Thanks for taking a look
> 
> I think this can be simplified a lot by using the [`vscode.open` command](https://code.visualstudio.com/api/references/commands) to show the image. That will open the image using VS Code's default editor. You can call this from the VS Code extension side with the uri of the image
> 
> To simplify things, this should only apply to images that are part of the workspace

Got it, lemme try

**Comment by Gobinath-B** [comment]
> Thanks for taking a look
> 
> I think this can be simplified a lot by using the [`vscode.open` command](https://code.visualstudio.com/api/references/commands) to show the image. That will open the image using VS Code's default editor. You can call this from the VS Code extension side with the uri of the image
> 
> To simplify things, this should only apply to images that are part of the workspace

hi @mjbvz , thanks for your guidance. I tried with vscode.open command. Now i can able to open a new tab with Image title, but it open to the side. and i'm little confusing with the image uri passing to the command. Can you give a guidance to overcome this.

**Comment by Gobinath-B** [comment]
I made changes , but the image is not rendered in the new tab. in console `An iframe which has both allow-scripts and allow-same-origin for its sandbox attribute can escape its sandboxing.", source: vscode-file://vscode-app/e:/Open%20Source/vscode/out/vs/workbench/contrib/webview/browser/webviewElement.js (326)` was shown. any ideas to fix this @mjbvz @Tyriar

File: extensions/markdown-language-features/preview-src/index.ts [hunk 1]
```diff
@@ -168,6 +168,16 @@ async function copyImage(image: HTMLImageElement, retries = 5) {
 	}
 }
 
+async function openImage(image: HTMLImageElement) {
```
Comment (mjbvz) [inline extensions/markdown-language-features/preview-src/index.ts:171]: [nit] Doesn't need to be async and too many extra spaces added afterwards

File: extensions/markdown-language-features/src/preview/preview.ts [hunk 2]
```diff
@@ -524,6 +525,11 @@ export class StaticMarkdownPreview extends Disposable implements IManagedMarkdow
 		});
 	}
 
+	async openImage() {
+		const uri = this.resource.toString;
```
Comment (mjbvz) [inline extensions/markdown-language-features/src/preview/preview.ts:529]: Should open the `resource` argument instead

File: src/vs/workbench/contrib/chat/browser/chatListRenderer.ts [hunk 3]
```diff
@@ -1239,17 +1239,20 @@ class ContentReferencesListRenderer implements IListRenderer<IChatContentReferen
 		const reference = data.reference;
 		templateData.label.element.style.display = 'flex';
 		if ('variableName' in reference) {
+			const variable = this.chatVariablesService.getVariable(reference.variableName);
```
Comment (mjbvz) [inline src/vs/workbench/contrib/chat/browser/chatListRenderer.ts:1242]: Looks like some other changes got included by mistake. Try merging your branch back into `main` and fixing any merge conflicts

File: extensions/markdown-language-features/package.json [hunk 4]
```diff
@@ -191,6 +196,10 @@
         {
           "command": "_markdown.copyImage",
           "when": "webviewId == 'markdown.preview' && webviewSection == 'image'"
+        },
+        {
+          "command": "_markdown.openImage",
```
Comment (mjbvz) [inline extensions/markdown-language-features/package.json:201]: This command should likely only be enabled for images that can be opened locally. For example, images that are part of the workspace

For this, you'll need to add a new `webviewSection` called something like `localImage` that is only set if the image is local. Take a look at how the contexts are set here:

https://github.com/microsoft/vscode/blob/5ee455116f5399293a854129299ed917ee8d4665/extensions/markdown-language-features/preview-src/index.ts#L129

File: extensions/markdown-language-features/preview-src/index.ts (hunk 1 above)
Comment (mjbvz) [reply extensions/markdown-language-features/preview-src/index.ts:171]: Also add some extra checks to make sure we only fire this for local images

File: extensions/markdown-language-features/src/commands/openImage.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+export class OpenImageCommand implements Command {
+	public readonly id = '_markdown.openImage';
+	public constructor(
+		private readonly _webviewManager: MarkdownPreviewManager,
+		private _currentPanel: vscode.WebviewPanel | undefined = undefined
+	) { }
+
+	public execute(args: { src: string; id: string }) {
+		const columnToShowIn = vscode.window.activeTextEditor
```
Comment (mjbvz) [inline extensions/markdown-language-features/src/commands/openImage.ts:18]: Can this be more similar to `CopyImageCommand`? I don't think the extra logic calling `reveal`/`create` is needed

**Review by mjbvz — COMMENTED** [review]
Added a few comments. However I think you can try this flow to simplify the change:

1. In the webview, update `data-vscode-context` to storage a value called `imageSrc` for local images: https://github.com/microsoft/vscode/blob/5ee455116f5399293a854129299ed917ee8d4665/extensions/markdown-language-features/preview-src/index.ts#L129

1. In `_markdown.openImage`, call `vscode.open` with the `imageSrc` that is passed to it. You may also need to look up the webview the command called from to get the view column to open the image in. You can use `findPreview` for that, similar to `CopyImageCommand`

**Comment by mjbvz** [comment]
Closing as it looks like this PR is out of date. Left a some comments earlier if you'd like to revisit it

**Pull request closed without merging**